auth/src/main/java/com/firebase/ui/auth/util/data/PrivacyDisclosureUtils.java

```diff
@@ -0,0 +1,90 @@
+package com.firebase.ui.auth.util.data;
+
+import android.content.Context;
+import android.support.annotation.RestrictTo;
+import android.support.annotation.StringRes;
+import android.widget.TextView;
+
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.data.model.FlowParameters;
+import com.firebase.ui.auth.util.ui.PreambleHandler;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class PrivacyDisclosureUtils {
+
+    private static final int NO_TOS_OR_PP = -1;
+    public static void setupTermsOfServiceAndPrivacyPolicyText(Context context,
+                                                               FlowParameters flowParameters,
+                                                               TextView termsText) {
+        PreambleHandler.setup(context,
+                flowParameters,
+                getGlobalTermsStringResource(flowParameters),
+                termsText);
+    }
+
+    public static void setupTermsOfServiceFooter(Context context,
+                                                 FlowParameters flowParameters,
+                                                 TextView footerText) {
+        PreambleHandler.setup(context,
+                flowParameters,
+                getGlobalTermsFooterStringResource(flowParameters),
+                footerText);
+    }
+
+    public static void setupTermsOfServiceAndPrivacyPolicySmsText(Context context,
+                                                                  FlowParameters flowParameters,
+                                                                  TextView termsText,
+                                                                  String multipleProviderFlowSmsMsg) {
+        if (flowParameters.isSingleProviderFlow()) {
+            PreambleHandler.setup(context,
+                    flowParameters,
+                    R.string.fui_verify_phone_number,
+                    getTermsSmsStringResource(flowParameters),
+                    termsText);
+        } else {
+            termsText.setText(multipleProviderFlowSmsMsg);
+        }
+    }
+
+    private static @StringRes
+    int getGlobalTermsStringResource(FlowParameters flowParameters) {
```

Comment:
Formatting nit (here and elsewhere) keep the whole method declaration up to the arg list on one line:

```java
private static @StringRes int getGlobalTermsStringResource(FlowParameters flowParameters) {
```

If you need to line break, do it after the `(`